pkg/runtime/depot.go

```diff
@@ -270,16 +273,44 @@ func (d *depot) Undeploy(id strfmt.UUID, relativeSrc, path string) error {
 	if !ok {
 		return errs.New("deployment for %s not found in depot", id)
 	}
-	deploy := sliceutils.Filter(deployments, func(d deployment) bool { return d.Path == path })
-	if len(deploy) != 1 {
+	deployments = sliceutils.Filter(deployments, func(d deployment) bool { return d.Path == path })
+	if len(deployments) != 1 {
 		return errs.New("no deployment found for %s in depot", path)
 	}
 
+	// Determine if there are any files provided by another artifact that will need to be re-linked
+	// or re-copied after this artifact is uninstalled.
+	relinks, err := d.getSharedFilesToReLink(id, relativeSrc, path)
```

Comment:
We shouldn't use the term "link" here, because they may be installed via copy. Suggest using "redeploys" and "getSharedFilesToRedeploy".